step-automation-packages/step-automation-packages-client/src/main/java/step/automation/packages/client/RemoteAutomationPackageClientImpl.java

```diff
@@ -84,6 +84,22 @@ public void deleteAutomationPackage(String packageId) throws AutomationPackageCl
         executeAutomationPackageClientRequest(() -> builder.delete(Void.class));
     }
 
+    @Override
+    public AutomationPackageUpdateResult createOrUpdateAutomationPackageLibrary(AutomationPackageSource librarySource, String managedLibraryName) throws AutomationPackageClientException {
+        MultiPart multiPart = new MultiPart();
+        multiPart.setMediaType(MediaType.MULTIPART_FORM_DATA_TYPE);
+        if (librarySource != null && librarySource.getFile() != null) {
```

Comment:
Just a comment, not a request to change: for this kinds of situations, you can also use

Optional.ofNullable(librarySource).map(...::getFile).ifPresent(file -> ...)

And if used multiple times, you can keep the first part as a variable:
var maybeLibrarySource = Optional.ofNullable(librarySource)

Depending on the situation, this may be more readable and efficient, but here, it's perfectly fine as is.